core/src/test/scala/org/apache/spark/executor/CoarseGrainedExecutorBackendSuite.scala

```diff
@@ -535,6 +538,40 @@ class CoarseGrainedExecutorBackendSuite extends SparkFunSuite
     }
   }
 
+  /**
+   * A fatal error occurred when [[Executor]] was initialized, this should be caught by
+   * [[SparkUncaughtExceptionHandler]] and [[Executor]] can exit by itself.
+   */
+  test("SPARK-40320 Executor should exit when initialization failed for fatal error") {
+    new SparkUncaughtExceptionHandler
+    val conf = new SparkConf()
+      .setMaster("local-cluster[1, 1, 1024]")
+      .set(PLUGINS, Seq(classOf[FatalErrorPlugin].getName))
+      .setAppName("test")
+    sc = new SparkContext(conf)
+    val executorAddCounter = new AtomicInteger(0)
+    val executorRemovedCounter = new AtomicInteger(0)
+
+    val listener = new SparkListener() {
+      override def onExecutorAdded(executorAdded: SparkListenerExecutorAdded): Unit = {
+        executorAddCounter.getAndIncrement()
+      }
+
+      override def onExecutorRemoved(executorRemoved: SparkListenerExecutorRemoved): Unit = {
+        executorRemovedCounter.getAndIncrement()
+      }
+    }
+    try {
+      sc.addSparkListener(listener)
+      eventually(timeout(60.seconds)) {
```

Comment:
Do we want to run this for 60s ? Perhaps, reduce it to 10s ?
Thoughts @Ngone51 ?

Reply:
This test takes about 5s. I make the timeout time longer here to hope that the UT will be more stable. When the machine load is heavy, the time-consuming time may become longer. 
I don't know if I need to worry about these, how about the modification to 20s.